Core/Battler.gd

```diff
@@ -110,22 +111,7 @@ func _try_player_play_card_on_entity(entity: Entity) -> void:
 		var can_play: bool = queued_card_data.can_play_card(PlayerManager.player, entity)
 		
 		if can_play:
-			CardManager.card_container.play_card([entity])
-
-
-func get_all_targets(application_type : Enums.ApplicationType) -> Array[Entity]:
-	var all_target : Array[Entity]
-	
-	match application_type:
-		Enums.ApplicationType.ALL:
-			all_target = _enemy_list
-			all_target += [PlayerManager.player]
-		Enums.ApplicationType.ENEMY_ONLY:
-			all_target = _enemy_list
-		Enums.ApplicationType.FRIENDLY_ONLY:
-			all_target = [PlayerManager.player]
-			
-	return all_target
+			CardManager.card_container.play_card(entity)
 
   
 func _handle_enemy_deaths() -> void:
```

Comment:
Should we remove the enemy from `EnemyManager.enemy_list` too when they die?

Reply:
list is passed by reference, meaning if you remove an enemy from the original list, it will also be removed in the copy
i can add assert in the tests to make sure the two lists stay equal if you would like

Reply:
Oh cool, didn’t know it did that. An assert would be good

Reply:
i added a test for that